src/Tests/Backend/One/OneToolchain.test.ts

```diff
@@ -15,39 +15,187 @@
+    test('NEG: check invalid version format without numbers', function() {
+      const oneCompiler = new OneCompiler();
+      assert.throws(() => oneCompiler.parseVersion('a.b.c'));
+    });
+    test('NEG: check invalid version format with too many numbers', function() {
+      const oneCompiler = new OneCompiler();
+      assert.throws(() => oneCompiler.parseVersion('1.2.3.4'));
+    });
+    test('NEG: check invalid version format with empty string', function() {
+      const oneCompiler = new OneCompiler();
+      assert.throws(() => oneCompiler.parseVersion(''));
+    });
+  });
 
-    // suite('#getInstalledToolchains', function() {
+  suite('#getToolchains', function() {
+    test('get toolchain list', function() {
+      // No positive test for get toolchain list
+      // To test this test case, prerequisites() function should be run first.
+      // However this function needs root permission so it couldn't be executed.
+      const oneCompiler = new OneCompiler();
+      assert.isDefined(oneCompiler.getToolchains);
+    });
 
-    // });
+    test('return empty toolchain array when count is 0', function() {
+      const oneCompiler = new OneCompiler();
+      const toolchainType = 'latest';
+      const start = 0;
+      const count = 0;
+      assert.deepStrictEqual(oneCompiler.getToolchains(toolchainType, start, count), []);
+    });
 
-    // suite('#prerequisitesForGetToolchains', function() {
+    test('NEG: request wrong toolchain type', function() {
+      const oneCompiler = new OneCompiler();
+      const dummyToolchainType = 'dummy';
+      const start = 0;
+      const count = 1;
+      assert.throws(() => oneCompiler.getToolchains(dummyToolchainType, start, count));
+    });
 
-    // });
+    test('NEG: request wrong start number', function() {
+      const oneCompiler = new OneCompiler();
+      const toolchainType = 'latest';
+      const start = -1;
+      const count = 1;
+      assert.throws(() => oneCompiler.getToolchains(toolchainType, start, count));
+    });
+
+    test('NEG: request wrong count number', function() {
+      const oneCompiler = new OneCompiler();
+      const toolchainType = 'latest';
+      const start = 0;
+      const count = -1;
+      assert.throws(() => oneCompiler.getToolchains(toolchainType, start, count));
+    });
+  });
+
+  suite('#getInstalledToolchains', function() {
+    test('get toolchain list', function() {
+      // No positive test for get toolchain list
+      // To test this test case, prerequisites() function should be run first.
+      // However this function needs root permission so it couldn't be executed.
+      const oneCompiler = new OneCompiler();
+      assert.isDefined(oneCompiler.getInstalledToolchains);
+    });
+
+    test('returns an empty installed toolchain list', function() {
+      const oneCompiler = new OneCompiler();
+      const toolchainType = 'latest';
+      assert.deepStrictEqual(oneCompiler.getInstalledToolchains(toolchainType), []);
+    });
```

Comment:
This tc result operates differently depending on the host environment. How about just checking if the function has been declared?

Reply:
When we use `isDefined`, it's like the test above. So we decided to remove this test.